src/main/resources/META-INF/jqassistant-rules/injection/jakartaee-injection.xml

```diff
@@ -21,4 +21,20 @@
         ]]></cypher>
     </concept>
 
+    <concept id="jakarta-injection:Resource">
+        <providesConcept refId="jee-injection:Resource"/>
+        <description>Labels all types, methods and fields annotated by @jakarta.annotation.Resource with "JEE" and "Resource".
+        </description>
+        <cypher><![CDATA[
+            MATCH
+              (e)-[:ANNOTATED_BY]->()-[:OF_TYPE]->(resourceAnnotationType:Type)
+            WHERE
+              (e:Type or e:Method or e:Field) AND resourceAnnotationType.fqn="jakarta.annotation.Resource"
+            SET
+              e:JEE:Resource
+            RETURN
+              e AS Resource
```

Comment:
Would it make sense to return the declaring type as well?
As the annotated elements can be types and members (i.e. fields and methods) the match clause would need to include both, e.g.

```
MATCH
  (t:Type)-[:DECLARES*0..1]->(e),
  (e)-[:ANNOTATED_BY]->()-[:OF_TYPE]->(resourceAnnotationType:Type)
...
RETURN
  t, e
ORDER BY
  t.fqn, e.name
```

This would apply to the other concepts as well.